[Android] Converting dimensions to pixels in popup renderer

geskill opened the pull request:
<!-- 
Hey there friend! First of all, thank you so much for this PR!
Some things that you should be aware of before opening this amazing PR❣

1. Please check if you are targeting the correct branch, we work on two branches and you should target the:
**main**: If you are fixing a bug and this bug doesn't change our public APIs
**develop**: If you are fixing a bug that introduces an API change, for example when you are implementing a new feature.
If you are not sure what branch target, ping one of the team directly or ask in the PR/issue and we will get back to you ASAP!

2. Before doing a lot of work, please check if there's an open issue for this chang, If not, please open an issue first so we can discuss upfront.

Also make sure you've read our Contribution guide here: https://github.com/xamarin/XamarinCommunityToolkit/blob/pj/update-pr-template/CONTRIBUTING.md
-->

### Description of Change ###

<!-- Describe your changes here. -->

### Bugs Fixed ###
<!-- Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR. -->

- Fixes #

### API Changes ###

None.

### Behavioral Changes ###

None.

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

Cfun1 commented on src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/Android/PopupRenderer.android.cs:
```diff
@@ -165,30 +171,30 @@ void SetSize()
 				}
 
-				if (Element.Content.HeightRequest > -1)
+				if (realContentHeight > -1)
```
Not sure if `realContentHeight ` will work fine as it was calculated based on `Element.Size.Width`, while here it was `Element.Content.HeightRequest`

Cfun1 commented:
It looks good, tried it on the sample.

pictos commented:
@ahoefling can you take a look at this one, when it's ready for review?

Cfun1 commented:
@pictos can you rerun the checks from here, I think it an issue in Azure not the code

pictos commented:
@Cfun1 I don't believe that I have permission to trigger the build. The easiest way to do it is the author of the PR send a dumb commit

ahoefling commented:
@pictos can you assign me as a reviewer I'll take a look tomorrow morning. Checked from my phone and this appears to make sense

pictos commented:
@ahoefling for some reason I can't select your name :( 
@jfversluis can you assign Andrew as a reviewer of this PR?

pictos commented:
@ahoefling I can't assign you as a reviewer, but I can assign this PR to you, not sure if this makes GitHub notify you whatever this PR changes.

ahoefling reviewed (changes requested):
The implementation looks great to me! Thank you so much for this because the Popup Size will be similar to iOS and UWP making any sizing math much easier.

When I first ran this in the sample app the popups were being drawn off screen because they were not updated. Please go back and update the sample code. Check all the popup.xaml files for a size and update accordingly. To make this easier there is a `PopupSize.cs` class with constants you can update. That isn't used in all popups, so you'll need to audit the rest.

Here is what it looks like unaltered
![image](https://user-images.githubusercontent.com/17751436/107234646-e0885d00-69f1-11eb-979d-0be4b6132647.png)

When I updated the size to `300x300` it looked like it did before. I would push the changes but I am not an official repo maintainer, so I'll need you or another maintainer to get in there and make this change

ahoefling commented:
This PR is missing the `Popup` label, can we have a maintainer add it

geskill commented:
I updated the `PopupSize.cs` definitions, like somehow mentioned here https://github.com/xamarin/XamarinCommunityToolkit/pull/839#issuecomment-774756295 .. now we have relative to screen size and fixed size examples.

jfversluis commented:
Let me know when this is ready to go according to you @ahoefling !

ahoefling commented on samples/XCT.Sample/Pages/Views/Popups/PopupSize.cs:
```diff
@@ -1,13 +1,18 @@
+		public static Size Small => new Size(300, 300);
+
+		// examples for relative to screen sizes
+		public static Size Medium => new Size(0.7 * (DeviceDisplay.MainDisplayInfo.Width / DeviceDisplay.MainDisplayInfo.Density), 0.6 * (DeviceDisplay.MainDisplayInfo.Height / DeviceDisplay.MainDisplayInfo.Density));
```
I like this, great idea!

ahoefling reviewed (commented):
This looks good to me and I think we are safe to merge. I am going to download the updates later tonight to do a final check before I approve the PR

ahoefling reviewed (approved):
Just downloaded the code and everything looks good.
